x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -417,35 +419,50 @@ private Map<String, List<FieldInferenceRequest>> createFieldInferenceRequests(Bu
                     String field = entry.getName();
                     String inferenceId = entry.getInferenceId();
                     Object inferenceResult = inferenceMap.remove(field);
-                    var value = XContentMapValues.extractValue(field, docMap);
-                    if (value == null) {
-                        if (inferenceResult != null) {
+                    for (var sourceField : entry.getSourceFields()) {
+                        var value = XContentMapValues.extractValue(sourceField, docMap);
+                        if (value == null) {
+                            if (isUpdateRequest) {
+                                addInferenceResponseFailure(
```

Comment:
This looks good, except for one edge case: Say that a user wants to use an update request to delete a field from a document. Traditionally, they would do this by explicitly setting the field value to `null`. How do we tell the difference between the `null` we get when the user didn't set a value and the `null` we get when the user explicitly set it?

IMO since we have a way to support partial updates in the near future (chunk source attribution), we could just live with this edge case for now.